### ISSUE TITLE
module promox is broken with Proxmox VE 4.0

### ISSUE THREAD

**Issue by bmillemathias** [ansible/ansible-modules-extras#1217]
AFAIK proxmox dropped support of openvz container in favor of lxc in the latest release (4.0 as of today), but module proxmox uses the openvz type container.
Any attempt for me gave an error 501, confirmed by the pveproxy log `"POST /api2/json/nodes/cactus/openvz HTTP/1.1" 501 130"`

I tried to replace any `openvz` call by a `lxc` call but it doesn't suffice to make the module work.
it returns such error

``` bash
failed: [localhost] => {"failed": true}
msg: creation of VM 102 failed with exception: 400 Bad Request
```

it is confirmed by the entry I got in the pveproxy log

**Comment by bmillemathias** [comment]
I'm still stuck trying to debug what the server does expect but I cannot find information.
I'd appreciate any advise to get more information on server or client side.

Thansk

**Comment by robynbergeron** [comment]
@bmillemathias -- thanks for filing this bug report.

@undergreen -- thanks in advance for taking a look at this bug report and for your ongoing work in maintaining this module.

**Comment by robynbergeron** [comment]
[module: cloud/misc/proxmox.py]

**Comment by bmillemathias** [comment]
The openvz to lxc migration patch is located at https://github.com/bmillemathias/ansible-modules-extras/commit/2ee4bc08c910d866bf8d8f862bd84e61ef6da3a8

**Comment by ThomasLamprecht** [comment]
http://pve.proxmox.com/wiki/Manual:_pct look at the create command for the changed things, it should be do-able to adapt this for PVE4, using if clauses do differ between PVE2/3 and PVE4 should suffice to not break compatibility.
I haven't done anything with ansible, but will try to give it a look, but no promises :)

**Comment by vialcollet** [comment]
Hi there.
Has someone started to look at this a bit further?

**Comment by adriagalin** [comment]
@bmillemathias @vialcollet this module was updated recently with lxc support. I have fixed a bug on netif param [PR 2064](https://github.com/ansible/ansible-modules-extras/pull/2064). I think this changes will be integrated in the next release. Meanwhile, you can download proxmox.py and put it in your library folder.

**Comment by resmo** [comment]
Closing this issue as #2064 has been merged and is marked to fix this issue

**Comment by bmillemathias** [comment]
Hi
Is there any reason that would prevent this fix and PR 2064 from being backported to 2.0.x ?

Thanks
